### DISCUSSION TITLE
How to compile library for ARM device

### DISCUSSION THREAD

**Discussion by ubx-luis** [protobuf-c/protobuf-c#603] [Q&A] [answered]
Has anyone have added protobuf to an embedded device?, I have generated the *.c/.h source code with protobuf definitions from my *.proto, but can't link the libraries as my IDE says:

attempt to link an object file that is not built for Advanced RISC Machines ARM ("lib/libprotobuf-c.a<protobuf-c.o>" = AMD x86-64 architecture)

I tried compiling google protobuf with the configuration set like: ./configure --host=arm, then did the same when configuring protobuf-c
but the libraries generated still seems to be for AMD... any idea?

**Comment by lipnitsk** [comment, marked as answer]
If you are having issues with autotools, you may have to just manually compile protobuf-c.c, maybe with minor modifications to fit your platform and the build toolchain (i.e. redefine `malloc` or `assert`, define `WORDS_BIGENDIAN`, etc.). I'm sure protobuf-c has been used on many embedded platforms. The library is simple to build since it consists of only one source file and one header file.
